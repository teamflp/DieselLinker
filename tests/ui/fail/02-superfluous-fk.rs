use diesel_linker::relation;

#[derive(Debug)]
struct User;

#[derive(Debug)]
#[relation(
    relation_type = "one_to_one",
    model = "User",
    fk = "user_id", // This is not used for a one_to_one relation
    backend = "sqlite"
)]
struct UserProfile;

fn main() {}
