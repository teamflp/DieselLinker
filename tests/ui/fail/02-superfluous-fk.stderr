error: `fk` is not used for this relation type. The foreign key is defined on the child model with `#[diesel(belongs_to(...))]`.
  --> tests/ui/fail/02-superfluous-fk.rs:10:5
   |
10 |     fk = "user_id", // This is not used for a one_to_one relation
   |     ^^

error[E0412]: cannot find type `UserProfile` in this scope
 --> tests/ui/fail/02-superfluous-fk.rs:6:10
  |
6 | #[derive(Debug)]
  |          ^^^^^ not found in this scope
